Cover Blazor debug with firewall config

guardrex opened the pull request:
Fixes #20411

Not sure about the language ... u might need to suggest alternate text.

* Is there a specific port that we can say to open?
* Are there one or several ports?

captainsafia commented on aspnetcore/blazor/debug.md:
```diff
@@ -332,6 +334,7 @@ If you're running into errors, the following tips may help:
 * If your environment uses an HTTP proxy, make sure that `localhost` is included in the proxy bypass settings. This can be done by setting the `NO_PROXY` environment variable in either:
   * The `launchSettings.json` file for the project.
   * At the user or system environment variables level for it to apply to all apps. When using an environment variable, restart Visual Studio for the change to take effect.
+* Ensure that no firewalls or proxies block communication with the debug proxy. If the [Windows Firewall](/windows/security/threat-protection/windows-firewall/windows-firewall-with-advanced-security) is blocking the communication, [create an inbound port rule](/windows/security/threat-protection/windows-firewall/create-an-inbound-port-rule) for any blocked ports.
```
This one is gonna be tricky because we intentionally use random ports when communicating with the mono proxy. If possible, I would recommend instructing users to allow "NodeJS" processes to communicate over private networks.

guardrex commented:
@captainsafia .... Needs work but hopefully is headed in the right direction. I think we need a section (added on the last commit). :ear:

When it comes to a firewall config, especially that permits a whole service to communicate freely, my 🚨 goes off 😨. Let me cover myself and Aquent with a quick ping to @blowdart.

@blowdart TL;DR ... We need to indicate for Blazor WASM debugging that a firewall rule might be required that permits free communication with the `NodeJs` service. The draft/WIP language that I just added goes like this ...

> \#\# Firewall configuration
>
> If a firewall blocks communication with the debug proxy, create a firewall exception rule that permits communication between the browser and the \`NodeJs\` service.
>
> For \[Windows Firewall](/windows/security/threat-protection/windows-firewall/windows-firewall-with-advanced-security) configuration guidance, see \[Create an Inbound Program or Service Rule](/windows/security/threat-protection/windows-firewall/create-an-inbound-program-or-service-rule). For more information, see \[Windows Defender Firewall with Advanced Security](/windows/security/threat-protection/windows-firewall/windows-firewall-with-advanced-security) and related articles in the Windows Firewall documentation set.
>
> > \[!WARNING]
> > Modification of a firewall configuration must be made with care to avoid creating security vulnerablities. Carefully apply security guidance, follow best security practices, and respect warnings issued by the firewall's manufacturer.

... :ear: for your guidance language.

blowdart commented:
Oh the joy of firewalls becoming software blockers.

You should indicate that this is only advisable on developer machines and that it opens up the node server to any and all connections, depending on your firewall capabilities which may be risky depending on your network. If possible only allow it to listen on trusted or private networks.

guardrex commented:
Thanks @blowdart. I incorporated that language.

I also moved the WARNING up (before the paragraph with the links to the Windows Firewall docs) to help ensure that readers consume the guidance.

captainsafia reviewed (approved):
Looks good! Thanks for checking in with @blowdart wrt to security.

guardrex commented:
"**_In @blowdart we trust!_**" is printed on Microsoft 💵. :smile:
